client/src/components/restaurant/restaurant-register.js

```diff
@@ -0,0 +1,289 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Button, Form, Container, Alert,
+} from 'react-bootstrap';
+import '../../styles/restaurant-registration.css';
+import Api from '../../services/api';
+import MyBadge from '../shared/my-batch';
+
+class RestaurantRegistration extends Component {
+  constructor(props) {
+    super(props);
+    this.endpoint = '';
+    this.state = {
+      name: '',
+      email: '',
+      address: '',
+      time_from: '09:00',
+      time_to: '22:00',
+      description: '',
+      tables: 3,
+      isRegistered: false,
+      isBadRequestError: '',
+      isValid: false,
+    };
+    this.isValidName = true;
+    this.isValidEmail = true;
+    this.isValidAddress = true;
+    this.isValidWorkingTime = true;
+    this.isValidCountTables = true;
+    this.isValidDescription = true;
+    this.userID = localStorage.getItem('userID');
+    this.handleChange = this.handleChange.bind(this);
+    this.showRegistration = this.showRegistration.bind(this);
+  }
+
+  // eslint-disable-next-line react/sort-comp
+  isValidCheck() {
+    const { isValid } = this.state;
+    this.setState({ isValid: true });
+    setTimeout(() => {
+      this.setState({
+        isValid: false,
+      });
+    }, 3000);
+    return isValid;
+  }
+
+  // eslint-disable-next-line react/sort-comp
+  handleChange(event) {
+    this.setState({
+      [event.target.id]: event.target.value,
+    });
+  }
+
+  validateForm() {
+    const {
+      email, name, address, description, tables,
+    } = this.state;
+    return email.length > 3 && name.length > 2 && address.length > 1 && description.length > 2 && tables > 0;
+  }
+
+  componentDidMount() { this.validateForm(); }
+
+  validInput() {
+    const {
+      // eslint-disable-next-line camelcase
+      email, name, address, description, tables, time_from, time_to,
+    } = this.state;
+    if (email.length < 3) { this.isValidEmail = false; } else { this.isValidEmail = true; }
+    if (name.length < 2) { this.isValidName = false; } else { this.isValidName = true; }
+    if (address.length < 1) { this.isValidAddress = false; } else { this.isValidAddress = true; }
+    if (description.length < 2) { this.isValidDescription = false; } else { this.isValidDescription = true; }
+    // eslint-disable-next-line max-len
+    if (time_from.length < 1 && time_to.length < 1) { this.isValidWorkingTime = false; } else { this.isValidWorkingTime = true; }
+    if (tables < 0) { this.isValidCountTables = false; } else { this.isValidCountTables = true; }
+    return this.isValidName && this.isValidEmail && this.isValidAddress
+    && this.isValidCountTables && this.isValidDescription && this.isValidWorkingTime;
+  }
+
+  showRegistration() {
+    this.setState({ isRegistered: true });
+    setTimeout(() => {
```

Comment:
Could you remove all timeouts
React bootstrap has highlevel handlers for similar cases https://react-bootstrap.github.io/components/toasts/#autohide

Reply:
And it will fix this issue
![image](https://user-images.githubusercontent.com/6152066/81311166-b9c97e80-908d-11ea-8363-d9c76133ba0b.png)